### DISCUSSION TITLE
`SetDebug(true)` style output on erroneous response?

### DISCUSSION THREAD

**Discussion by raine** [go-resty/resty#535] [Q&A]
Hey, I have a function like follows that I use to convert >=400 status code response into go errors.

```go
func handleError(res *resty.Response, err error) (*resty.Response, error) {
	if err != nil {
		return res, err
	}
	if res.IsError() {
		log.Error().
			Str("url", res.Request.URL).
			Str("method", res.Request.Method).
			Interface("request", res.Request.Body).
			Interface("response", res.Body()).
			Send()
		return res, errors.Errorf("request failed: %s %s", res.Request.Method, res.Request.URL)
	}

	return res, nil
}
```

For failed requests, to help with debugging, I'd like to get the type of output that you see when you have `SetDebug(true)` enabled for a client. Even better if you can direct that output to a file. Does resty support something like this?

Thanks!

**Comment by moorereason** [comment]
The debug logger is [implemented as middleware in private functions](https://github.com/go-resty/resty/blob/master/middleware.go#L262-L320).  I doubt those will be exported as public functions.  You can probably reimplement the logging in your own middleware.
